addons/product/models/product_product.py

```diff
@@ -96,9 +96,9 @@ def _compute_can_image_variant_1024_be_zoomed(self):
     def _set_template_field(self, template_field, variant_field):
         for record in self:
             if (
-                # We are trying to remove a field from the variant even though it is already
-                # not set on the variant, remove it from the template instead.
-                (not record[template_field] and not record[variant_field])
+                # We are trying to remove the image field. Remove it from the
+                # template and from the variant.
+                not record[template_field]
                 # We are trying to add a field to the variant, but the template field is
                 # not set, write on the template instead.
                 or (record[template_field] and not record.product_tmpl_id[template_field])
```

Comment:
Due to the update of the condition, `record[template_field]` could be removed here. But maybe is it better to let it (easier for the comprehension)?